nodejs/eks/nodegroup.ts

```diff
@@ -709,6 +709,31 @@ export function createManagedNodeGroup(name: string, args: ManagedNodeGroupOptio
         roleArn = args.nodeRoleArn;
     }
 
+    // Check that the nodegroup role has been set on the cluster to
+    // ensure that the aws-auth configmap was properly formed.
+    const roleExistsInNodeAccess = pulumi.all([
+        core.instanceRoles,
+        roleArn,
+    ]).apply(([roles, rArn]) => {
+        // Map out the ARNs of all of the instanceRoles.
+        const roleArns = roles.map(role => {
+            return role.arn.apply(a => a);
+        });
+        // Try finding the nodeRole in the ARNs array.
+        return pulumi.all([
+            roleArns,
+            rArn,
+        ]).apply(([arns, arn]) => {
+            return arns.find(a => a === arn);
```

Comment:
It looks like this return type is expected to be boolean based on the variable name. `find` returns the actual element, so you would need to compare it to `-1`.

```ts
return arns.find(a => a === arn) !== -1
```

Reply:
The check for undefined is done below that codeblock:

`if (!exist)...`

On that note, it makes sense to rename the var to `role` vs `roleExistsInNodeAccess` to not insinuate a bool as I believe was the original design

https://github.com/pulumi/pulumi-eks/blob/0e93022bb6aa1b5967684034e9b05d9d9291b19e/nodejs/eks/nodegroup.ts#L712-L735

Reply:
Done. Renamed var to role